src/compiler/parser.ts

```diff
@@ -4616,10 +4629,10 @@ namespace ts {
                 factory.createConditionalExpression(
                     leftOperand,
                     questionToken,
-                    doOutsideOfContext(disallowInAndDecoratorContext, parseAssignmentExpressionOrHigher),
+                    doOutsideOfContext(disallowInAndDecoratorContext, () => parseAssignmentExpressionOrHigher(/*parsingConditionalTrueSide*/ true)),
                     colonToken = parseExpectedToken(SyntaxKind.ColonToken),
                     nodeIsPresent(colonToken)
-                        ? parseAssignmentExpressionOrHigher()
+                        ? parseAssignmentExpressionOrHigher(/*parsingConditionalTrueSide*/ false)

```

Comment:
This is `false` because I believe this to be unambiguous after the first colon, and one of the original test cases exhibits this.

(I could have omitted the parameter, but I'm being explicit in this one place for clarity.)

Reply:
This might be problematic in `a ? b ? c : (d) : e => f`

Reply:
Thanks for the test case. This should be parsed the same as:

```ts
a ? (b ? (c) : (d)) : (e => f)
```

I guess it's still ambiguous in the right side.